src/client/shader.cpp

```diff
@@ -682,6 +731,25 @@ ShaderInfo ShaderSource::generateShader(const std::string &name,
 
 	shaders_header << "#define FOG_START " << core::clamp(g_settings->getFloat("fog_start"), 0.0f, 0.99f) << "\n";
 
+	bool enable_shadows = g_settings->getBool("enable_dynamic_shadows");
+	if (drawtype == NDT_TORCHLIKE)
+		enable_shadows = false; // why?
```

Comment:
(this condition existed inside the shader before, I only moved it)
What's the purpose of this? To not apply shadows to light sources? If so, that should be done differently.

Reply:
Literally I have no idea why I put that in there...

Reply:
Okay, removed it then.